apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -579,18 +452,15 @@ impl Service<QueryPlannerRequest> for BridgeQueryPlanner {
                     )))
                 }
                 Ok(modified_query) => {
-                    let executable_document = modified_query
-                        .to_executable(schema)
-                        // Assume transformation creates a valid document: ignore conversion errors
-                        .unwrap_or_else(|invalid| invalid.partial);
+                    let executable_document =
+                        modified_query.to_executable_validate(schema).map_err(|e| {
```

Comment:
Validation is hopefully faster now but still, do we want to run it 2~3 times per query?

Reply:
Note also that some transformations do not produce valid documents today (an example in https://github.com/apollographql/router/pull/4938)

Reply:
We should probably keep this for now so we don't pass invalid documents to the query planner. When #4938 is fixed we can consider only validating if debug assertions are on and use `assume_valid` in prod?